library/core/src/slice/iter.rs

```diff
@@ -68,7 +68,7 @@ pub struct Iter<'a, T: 'a> {
     /// For non-ZSTs, the non-null pointer to the past-the-end element.
     ///
     /// For ZSTs, this is `ptr::invalid(len)`.
-    end: *const T,
+    end_or_len: *const T,
```

Comment:
I wish this could use a `union`, but it can't because that makes it able to store `undef`, which we don't want.